### ISSUE TITLE
Misleading error message "Cannot run in this PostgreSQL version" when postgres is specified as the authenticator user

### ISSUE THREAD

**Issue by GeraldScott** [begriffs/postgrest#576]
I tried to start an instance of PostgREST with the 'postgres' user in the connection string:

`[root@dbserver ~]# postgrest postgres://postgres:mypass@localhost:5432/demo1 -a postgres`

It started up with the usual messages but then fell over with a very misleading error message:

`WARNING, running in insecure mode, JWT secret is the default value`
`Listening on port 3000`
`postgrest: Cannot run in this PostgreSQL version, PostgREST needs at least 90300`

By contrast, it starts up with no problem if I remove postgres from the connection string:

`[root@dbserver ~]# postgrest postgres://localhost:5432/demo1 -a postgres `
`WARNING, running in insecure mode, JWT secret is the default value`
`Listening on port 3000`

**Comment by ruslantalpa** [comment]
it has noting to do with the parameters. Notice that you are connecting to two different servers (dbserver, gostorage). One of them has postgresql 9.3+ (the one thats working) the other has probably 9.2

**Comment by GeraldScott** [comment]
Oops, well spotted. I blundered when I tried to anonymize my server name and forgot the second entry. 

It is actually one server. PostgREST and PostgreSQL are on the same box. 

Something is causing an error which generates `Cannot run in this PostgreSQL version, PostgREST needs at least 90300`, despite the fact that I am running PostgreSQL 9.2.15. 

I see it comes from these lines in Main.hs:
`  result <- P.use pool $ do
    supported <- isServerVersionSupported
    unless supported $ error (
      "Cannot run in this PostgreSQL version, PostgREST needs at least "
      <> show minimumPgVersion)
    getDbStructure (cs $ configSchema conf)`

But I don't know Haskell, sigh.
